### ISSUE TITLE
`# type: ignore` is not interpreted if it is not the first part of comment

### ISSUE THREAD

**Issue by reynoldsnlp** [python/mypy#8744]
Many QA tools make use of comments to ignore certain "problems" with code.

* `mypy` uses `# type: ignore`.
* `flake8` uses `# noqa` or `# noqa: <SPECIFIC CODE>`
* `coverage` uses `pragma: no cover`

I have found that `mypy` does not recognize such comments unless it is the first part of a comment. In the following code `# noqa  # type: ignore` is not recognized by `mypy`:

```bash
$ cat /tmp/a.py
def a(x: str, y: int) -> str:
    return f'{x}{y}'

print(a(2, 's'))  # noqa  # type: ignore

$ mypy /tmp/a.py
/tmp/a.py:4: error: Argument 1 to "a" has incompatible type "int"; expected "str"
/tmp/a.py:4: error: Argument 2 to "a" has incompatible type "str"; expected "int"
Found 2 errors in 1 file (checked 1 source file)
```

If the order of the comments is switched, then the ignore comment is recognized:

```bash
$ cat /tmp/a.py
def a(x: str, y: int) -> str:
    return f'{x}{y}'

print(a(2, 's'))  # type: ignore  # noqa

$ mypy /tmp/a.py
Success: no issues found in 1 source file
```

To enable using a combination of tools, it seems like `mypy` should recognize the comment regardless of the order.

**Comment by ethanhs** [comment]
This is documented: https://mypy.readthedocs.io/en/stable/common_issues.html#silencing-linters

I also don't think this will be changed, because it precludes commenting out a type ignore. Also the parser has been upstreamed to CPython, so this would probably be best reported there.

**Comment by reynoldsnlp** [comment]
@ethanhs Sorry I didn't find that in the docs. Apparently I need to hone my google-fu skiils.
